test/Passes/pass-tile-and-fuse-mha.mlir

```diff
@@ -0,0 +1,135 @@
+// CHECK-DAG: #[[MAP9:.+]] = affine_map<(d0, d1, d2, d3) -> (d0, d3)>
+
+// CHECK-LABEL: mha_tensorflow
+func.func @mha_tensorflow(%arg1: tensor<64x32x8x64xf32>,
+                           %arg0: tensor<64x32x8x64xf32>,
+                           %arg2: tensor<64x32x8x64xf32>) -> tensor<64x32x512xf32> {
+  // CHECK: %[[C64:.+]] = arith.constant 64 : index
+  %cst_3 = arith.constant dense<4.0> : tensor<8x64x8x64xf32>
+  %cst_4 = arith.constant dense<2.0> : tensor<8x64x8x64xf32>
+  %cst_5 = arith.constant dense<1.0> : tensor<8x64x8x64xf32>
+  %cst_6 = arith.constant dense<3.0> : tensor<8x64x512xf32>
+
+  %cst_1 = arith.constant 0.0 : f32
+  %0 = tensor.empty() : tensor<64x32x8x64xf32>
+  
+  // CHECK: %{{.+}} = scf.forall (%{{.+}}) in (%[[C64]])
+  %fill = linalg.fill ins(%cst_1 : f32) outs(%0 : tensor<64x32x8x64xf32>) -> tensor<64x32x8x64xf32>
+  %2 = linalg.generic {
+    "__projection_V__",
+    indexing_maps = [#map, #map1, #map2], 
+    iterator_types = ["parallel", "parallel", "reduction", "reduction", "parallel", "parallel"]}
+    ins(%arg2, %cst_3 : tensor<64x32x8x64xf32>, tensor<8x64x8x64xf32>) outs(%fill : tensor<64x32x8x64xf32>) {
+    ^bb0(%in: f32, %in_8: f32, %out: f32):
+      %23 = arith.mulf %in, %in_8 : f32
+      %24 = arith.addf %out, %23 : f32
+      linalg.yield %24 : f32
+  } -> tensor<64x32x8x64xf32>
+  // CHECK: %{{.+}} = linalg.generic 
+  // CHECK-SAME:  indexing_maps = [#[[MAP]], #[[MAP1]], #[[MAP2]]], 
+  // CHECK-SAME:  iterator_types = ["parallel", "reduction", "reduction", "parallel", "parallel"]
+
+  %fill_1 = linalg.fill ins(%cst_1 : f32) outs(%0 : tensor<64x32x8x64xf32>) -> tensor<64x32x8x64xf32>
+  %3 = linalg.generic {
+    "__projection_Q__",
+    indexing_maps = [#map, #map1, #map2], 
+    iterator_types = ["parallel", "parallel", "reduction", "reduction", "parallel", "parallel"]}
+    ins(%arg0, %cst_4 : tensor<64x32x8x64xf32>, tensor<8x64x8x64xf32>) outs(%fill_1 : tensor<64x32x8x64xf32>) {
+    ^bb0(%in: f32, %in_8: f32, %out: f32):
+      %23 = arith.mulf %in, %in_8 : f32
+      %24 = arith.addf %out, %23 : f32
+      linalg.yield %24 : f32
+  } -> tensor<64x32x8x64xf32>
+  // CHECK: %{{.+}} = linalg.generic
+  // CHECK-SAME: indexing_maps = [#[[MAP]], #[[MAP1]], #[[MAP2]]]
+  // CHECK-SAME:  iterator_types = ["parallel", "reduction", "reduction", "parallel", "parallel"]
+
+  %fill_2 = linalg.fill ins(%cst_1 : f32) outs(%0 : tensor<64x32x8x64xf32>) -> tensor<64x32x8x64xf32>
+  %5 = linalg.generic {
+    "__projection_K__",
+    indexing_maps = [#map, #map1, #map2], 
+    iterator_types = ["parallel", "parallel", "reduction", "reduction", "parallel", "parallel"]}
+    ins(%arg1, %cst_5 : tensor<64x32x8x64xf32>, tensor<8x64x8x64xf32>) outs(%fill_2 : tensor<64x32x8x64xf32>) {
+    ^bb0(%in: f32, %in_8: f32, %out: f32):
+      %23 = arith.mulf %in, %in_8 : f32
+      %24 = arith.addf %out, %23 : f32
+      linalg.yield %24 : f32
+  } -> tensor<64x32x8x64xf32>
+  // This is __Q_times_K__
```

Comment:
If the pass moves code around, it's easier to write the original code in the order the pass will do, so that the check lines make more sense when we read